balance_stap.c

```diff
@@ -0,0 +1,42 @@
+#include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
+
+void balance(char* str)
+{
+    // флаг для вложенности
+    bool flag = true;
+
+    // коэф для проверки вложености скобок во время пробежки по строке(если коэф меньше 0, вложенность нарушена)
+    // открывающая скобка имеет коэф +1
+    // закрывающая скобка имеет коэф -1
+    int count = 0;
+    for (int i = 0; i < strlen(str); i++) {
+        if (str[i] == '(') {
+            count++;
+
+        } else if (str[i] == ')') {
+            count -= 1;
```

Comment:
Раз уж на то пошло, можно и так
```suggestion
            count--;
```